galley/pkg/crd/validation/config.go

```diff
@@ -23,26 +23,68 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 	"reflect"
+	"sync"
+	"time"
 
 	"github.com/ghodss/yaml"
+	"github.com/howeyc/fsnotify"
 	"k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	clientset "k8s.io/client-go/kubernetes"
 	admissionregistration "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
 	"k8s.io/client-go/tools/cache"
 
+	"istio.io/istio/pkg/kube"
+
 	"istio.io/istio/pkg/log"
 )
 
 var scope = log.RegisterScope("validation", "CRD validation debugging", 0)
 
+type createInformerWebhookSource func(cl clientset.Interface, name string) cache.ListerWatcher
+
+var (
+	defaultCreateInformerWebhookSource = func(cl clientset.Interface, name string) cache.ListerWatcher {
+		return cache.NewListWatchFromClient(
+			cl.AdmissionregistrationV1beta1().RESTClient(),
+			"validatingwebhookconfigurations",
+			"",
+			fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", name)))
+	}
+)
+
+// WebhookConfig implements the validating admission webhook for validating Istio configuration.
+type WebhookConfig struct {
+	mu                            sync.RWMutex
+	cert                          *tls.Certificate
```

Comment:
Some of these are coming from WebhookPArameters, which is effectively immutable. This should get a handle/copy of WebhookParameters and avoid duplicating those fields.

Reply:
fixed.